File: src/main/java/workshop/part2/supervisor/VettingSupervisor.java
package workshop.part2.supervisor;

import akka.actor.AbstractActor;

public class VettingSupervisor extends AbstractActor {

    private final VettingActorFactory vettingActorFactory;

    public VettingSupervisor(VettingActorFactory vettingActorFactory) {
        this.vettingActorFactory = vettingActorFactory;
    }

    @Override
    // Spawn a new VettingSubActor for every ad, delegate it, and spoof sender
    public Receive createReceive() {
        return receiveBuilder()
            .build();
    }
}

File: src/main/kotlin/workshop/part2/supervisor/VettingSupervisor.kt
package workshop.part2.supervisor

import akka.actor.AbstractActor

class VettingSupervisor internal constructor(private val vettingActorFactory: VettingActorFactory) : AbstractActor() {
    override fun createReceive(): Receive {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

}
